### ISSUE TITLE
Populating struct_fields with a form 0 

### ISSUE THREAD

**Issue by mbromley-voxel** [cosmocode/dokuwiki-plugin-struct#389]
I've setup a form to populate a struct schema on a new page.  The form works, the page is created, and the struct fields are populated as I would expect.  However, after submitting the form I get the following error above the link for the new page that was created:

Warning: preg_split() expects parameter 2 to be string, array given in /home/voxeli5/public_html/wiki/inc/parserutils.php on line 287

Warning: Invalid argument supplied for foreach() in /home/voxeli5/public_html/wiki/inc/parserutils.php on line 287

I opened the file where the error is and it seems to be this call that is failing:

`    // filter by $key
    foreach(preg_split('/\s+/', $key, 2, PREG_SPLIT_NO_EMPTY) as $cur_key) {
        if (!isset($val[$cur_key])) {
            return null;
        }`

**Comment by Juergen-aus-Koeln** [comment]
I have inserted a type conversion "(string)$key" into the file parserutils.php:
```
    // filter by $key
    settype($key, 'string');
    foreach(preg_split('/\s+/', (string)$key, 2, PREG_SPLIT_NO_EMPTY) as $cur_key) {
        if (!isset($val[$cur_key])) {
            return null;
        }
        $val = $val[$cur_key];
    }
 ```
I don't know anything about programming in PHP, but for me it works at the moment.

**Comment by Juergen-aus-Koeln** [comment]
I have inserted a type conversion "(string)$key" into the file parserutils.php:
```
    // filter by $key
    settype($key, 'string');
    foreach(preg_split('/\s+/', (string)$key, 2, PREG_SPLIT_NO_EMPTY) as $cur_key) {
        if (!isset($val[$cur_key])) {
            return null;
        }
        $val = $val[$cur_key];
    }
 ```
I don't know much about programming in PHP, but for me it works at the moment.

**Comment by blakskyy** [comment]
I am having this issue too

```
Warning: preg_split() expects parameter 2 to be string, array given in parserutils.php on line 289

Warning: Invalid argument supplied for foreach() in /home/xxxxx/public_html/inc/parserutils.php on line 289
```
Release 2018-04-22a "Greebo"
Installed version: 2018-10-08

**Comment by annda** [comment]
This error seems to be coming from a third-party plugin (not even bureaucracy). @mbromley-voxel  @Juergen-aus-Koeln @blakskyy Can any of you debug the `$key` variable and post the offending non-string content here?
